File: BookFlix.Core/Helpers/PasswordHelper.cs
namespace BookFlix.Core.Helpers
{
    public static class PasswordHelper
    {
        public static bool IsStrongPassword(string password)
        {
            if (string.IsNullOrEmpty(password) || password.Length < 8)
                return false;
            bool hasUpper = false, hasLower = false, hasDigit = false, hasSpecial = false;
            foreach (char c in password)
            {
                if (char.IsUpper(c)) hasUpper = true;
                else if (char.IsLower(c)) hasLower = true;
                else if (char.IsDigit(c)) hasDigit = true;
                else if (!char.IsLetterOrDigit(c)) hasSpecial = true;
                if (hasUpper && hasLower && hasDigit && hasSpecial)
                    return true;
            }
            return false;
        }

        public static string HashPassword(string password)
        {
            return BCrypt.Net.BCrypt.HashPassword(password);
        }

        public static bool VerifyPassword(string password, string hashedPassword)
        {
            return BCrypt.Net.BCrypt.Verify(password, hashedPassword);
        }
    }
}

File: Frontend/bookflixUI/src/app/shared/helpers/password-helper.ts
export class PasswordHelper {
  public static IsStrongPassword(password: string): boolean {
    if (password.length < 8) return false;
    let hasUpper: boolean = false,
      hasLower: boolean = false,
      hasDigit: boolean = false,
      hasSpecial: boolean = false;

    for (const c of password) {
      if (/[A-Z]/.test(c)) hasUpper = true;
      else if (/[a-z]/.test(c)) hasLower = true;
      else if (/[0-9]/.test(c)) hasDigit = true;
      else hasSpecial = true;

      if (hasUpper && hasLower && hasDigit && hasSpecial) {
        return true;
      }
    }

    return false;
  }
}
